cli/cmd/deploy/deploy.go

```diff
@@ -358,6 +388,169 @@ func loginWithTelemetry(ctx context.Context, ch *cmdutil.Helper, redirectURL str
 	return nil
 }
 
+func createGithubRepoFlow(ctx context.Context, ch *cmdutil.Helper, localGitPath string) error {
+	// Get the admin client
+	c, err := ch.Client()
+	if err != nil {
+		return err
+	}
+
+	res, err := c.GetGithubUserStatus(ctx, &adminv1.GetGithubUserStatusRequest{})
+	if err != nil {
+		return err
+	}
+
+	if !res.HasAccess {
+		ch.Telemetry(ctx).RecordBehavioralLegacy(activity.BehavioralEventGithubConnectedStart)
+
+		if res.GrantAccessUrl != "" {
+			// Print instructions to grant access
+			time.Sleep(3 * time.Second)
+			ch.Print("Open this URL in your browser to grant Rill access to Github:\n\n")
+			ch.Print("\t" + res.GrantAccessUrl + "\n\n")
+
+			// Open browser if possible
+			_ = browser.Open(res.GrantAccessUrl)
+		}
+	}
+
+	// Poll for permission granted
+	pollCtx, cancel := context.WithTimeout(ctx, pollTimeout)
+	defer cancel()
+	for {
+		select {
+		case <-pollCtx.Done():
+			return pollCtx.Err()
+		case <-time.After(pollInterval):
+			// Ready to check again.
+		}
+
+		// Poll for access to the Github's user account
+		pollRes, err := c.GetGithubUserStatus(ctx, &adminv1.GetGithubUserStatusRequest{})
+		if err != nil {
+			return err
+		}
+		if !pollRes.HasAccess {
+			// Sleep and poll again
+			continue
+		}
+
+		// Emit success telemetry
+		ch.Telemetry(ctx).RecordBehavioralLegacy(activity.BehavioralEventGithubConnectedSuccess)
+
+		repoOwner := pollRes.Account
+		if len(pollRes.Organizations) > 0 {
+			repoOwners := []string{pollRes.Account}
+			repoOwners = append(repoOwners, pollRes.Organizations...)
+			ch.Print("\nYou also have access to organization(s)\n\n")
+			repoOwner = cmdutil.SelectPrompt("Please chhose where to create the repository", repoOwners, pollRes.Account)
+		}
+		// create and verify
+		githubRepository, err := createGithubRepository(ctx, ch, pollRes, localGitPath, repoOwner)
+		if err != nil {
+			return err
+		}
+
+		// init git repo
+		repo, err := git.PlainInit(localGitPath, false)
+		if err != nil {
+			if !errors.Is(err, git.ErrRepositoryAlreadyExists) {
+				return fmt.Errorf("failed to init git repo: %w", err)
+			}
+			repo, err = git.PlainOpen(localGitPath)
+			if err != nil {
+				return fmt.Errorf("failed to open git repo: %w", err)
+			}
+		}
+
+		wt, err := repo.Worktree()
+		if err != nil {
+			return fmt.Errorf("failed to get worktree: %w", err)
+		}
+
```

Comment:
Any easy way to bail/prompt if the repo is bigger than e.g. 100 MB?

Reply:
I could find this command : `git count-objects -v -H` 
But the problem is this works only after files have been added into the repository. And adding bigger files itself takes lot of time (Took ~4 minutes to add a file of size ~4.5GB). 
Can't think of any other easy way.

Reply:
Okay, then let's like it like this for now